framework/src/main/java/org/apache/drill/test/framework/DrillTestJdbc.java

```diff
@@ -241,6 +245,73 @@ private void executeQuery(String query) throws IOException, SQLException {
     }
   }
 
+  private void executeLimitZeroQuery(String query) throws IOException, SQLException {
+	BufferedWriter writer = new BufferedWriter(new FileWriter(new File(outputFilename),true));
+    final boolean cancelQuery = rand.nextInt(100) < TestDriver.OPTIONS.cancelPercent;
+    CancelQuery c = null;
+    ResultSet resultSetA = null;
+    try {
+      Statement statement = connection.createStatement();
+      resultSetA = statement.executeQuery(query);      
+      if (cancelQuery) {
+        c = new CancelQuery(statement);
+        c.start();
+      }  
+    } finally {
+      if (cancelQuery) {
+        try {
+          c.join();
+          if (testStatus == TestStatus.CANCELED) {
+            if (resultSetA != null) {
+              resultSetA.close();
+            }
+            if (writer != null) {
+              writer.close();
+            }
+            return;
+          }
+        } catch (InterruptedException e) {
+          resultSetA.close();
+          writer.close();
+          return;
+        }
+      }
+    }
+    List columnLabelsA = new ArrayList<String>();
```

Comment:
You can use [variable shadowing](http://www.xyzws.com/javafaq/what-is-variable-hiding-and-shadowing/15), and name these local variables as is (resultSet, columnLabels, types, ...). To access the instance level variables use *this* (this.resultSet, this.columnLabels, ...).